src/bootstrap-table.js

```diff
@@ -46,7 +46,7 @@ class BootstrapTable {
       Utils.sprintf(`${buttonsPrefix}%s`, opts.iconSize)
     ].join(' ').trim()
 
-    this.buttons = Utils.calculateObjectValue(this, opts.buttons, [], [])
+    this.buttons = Utils.calculateObjectValue(opts, opts.buttons, undefined, [])
```

Comment:
This and Line 170 must be reverted, i also tested it but it cause some issue with other features i guess.